src/ORM/Table.php

```diff
@@ -1691,6 +1694,36 @@ public function query(): Query
         return new Query($this->getConnection(), $this);
     }
 
+    /**
+     * Creates a new insert query
+     *
+     * @return \Cake\ORM\Query\InsertQuery
+     */
+    public function insertQuery(): InsertQuery
+    {
+        return new InsertQuery($this->getConnection(), $this);
```

Comment:
I am wondering if we should have a `QueryFactory` for the ORM too.

Reply:
Would it cover `query()` and `subquery()` too?

Reply:
`query()` yes, not sure `subquery()` needs to be on the factory  method. We can instead add a public method on `Query` to disable aliasing which `Table::subquery()` can use.

Reply:
My thinking especially for `subquery()` was that it wouldn't be nice if I'd provide my own factory with custom/extended query classes, and subqueries would be an outlier that would stick to `\Cake\ORM\Query`.

Reply:
The query used by subquery() would be generated thorough the factory class, so it would use your custom query isn't it?

Reply:
If it would be changed to use a query generated through the factory, then yes. Just wanted to mention the basis of my initial concern to make sure we're not talking past each other.

Reply:
The current static method `Query::subquery()` would have to be dropped though. Hope that's okay.

Reply:
The method requires a table instance anyways, so `Query::subquery($table)` can easily be changed to `$table->subquery()`. Having an replaceable factory for all queries and reducing static dependencies in people's code, that's a win-win in my book.

Not sure what the point of the static method was initially (@othercorey ?), but if it was to be able set a protected property without introducing additional constructor arguments, then we could still do that and avoid public access by binding a closure to the query's scope, like:

```php
Closure::bind(
    fn () => $query->{'aliasingEnabled'} = false,
    null,
    $query
)();
```

Reply:
I have dropped `Query::subquery()`. The PR which adds this method mentioned it being useful for creating subqueries for join and CTE. Users can `Table::subquery()` for subqueries and `Query::with()` gets a db query through the connection https://github.com/cakephp/cakephp/blob/b2a6738ecc7027cd8a16d89d24f49e759b6fd36b/src/Database/Query.php#L430-L431